src/mechanics/mesh/MeshFem.cpp

```diff
@@ -100,17 +100,11 @@ Group<NodeSimple> MeshFem::NodesTotal(DofType d)
     Group<NodeSimple> group;
     for (auto& element : this->Elements)
     {
-        const ElementFem* dofElm;
-        try
+        if (element.Has(d))
```

Comment:
I personally would reverse the logic. Like return/continue as early as possible. This also reduces the level of indentation. This may be taste however.